packages/testcontainers/src/container-runtime/clients/container/docker-container-client.ts

```diff
@@ -201,34 +201,55 @@ export class DockerContainerClient implements ContainerClient {
       execOptions.User = opts.user;
     }
 
-    const chunks: string[] = [];
+    const outputChunks: string[] = [];
+    const stdoutChunks: string[] = [];
+    const stderrChunks: string[] = [];
+
     try {
       if (opts?.log) {
         log.debug(`Execing container with command "${command.join(" ")}"...`, { containerId: container.id });
       }
 
       const exec = await container.exec(execOptions);
-      const stream = await exec.start({ stdin: true, Detach: false, Tty: true });
-      if (opts?.log && execLog.enabled()) {
-        byline(stream).on("data", (line) => execLog.trace(line, { containerId: container.id }));
-      }
+      const stream = await exec.start({ stdin: true, Detach: false, Tty: false });
+
+      const stdoutStream = new PassThrough();
+      const stderrStream = new PassThrough();
+
+      this.dockerode.modem.demuxStream(stream, stdoutStream, stderrStream);
+
+      const processStream = (stream: Readable, chunks: string[]) => {
+        stream.on("data", (chunk) => {
+          chunks.push(chunk.toString());
+          outputChunks.push(chunk.toString());
+
+          if (opts?.log && execLog.enabled()) {
+            execLog.trace(chunk.toString(), { containerId: container.id });
+          }
+        });
+      };
+
+      processStream(stdoutStream, stdoutChunks);
+      processStream(stderrStream, stderrChunks);
 
       await new Promise((res, rej) => {
-        stream.on("data", (chunk) => chunks.push(chunk));
         stream.on("end", res);
         stream.on("error", rej);
       });
       stream.destroy();
 
       const inspectResult = await exec.inspect();
       const exitCode = inspectResult.ExitCode ?? -1;
-      const output = chunks.join("");
+      const output = outputChunks.join("");
+      const stdout = stdoutChunks.join("");
+      const stderr = stderrChunks.join("");
+
       if (opts?.log) {
         log.debug(`Execed container with command "${command.join(" ")}"...`, { containerId: container.id });
       }
-      return { output, exitCode };
+      return { output, stdout, stderr, exitCode };
     } catch (err) {
-      log.error(`Failed to exec container with command "${command.join(" ")}": ${err}: ${chunks.join("")}`, {
+      log.error(`Failed to exec container with command "${command.join(" ")}": ${err}: ${stderrChunks.join("")}`, {
```

Comment:
Perhaps this should still be `chunks`, I guess that an exec could fail without error output, in which case it would be useful to see all the output?